plugins/catalog-backend/src/service/AuthorizedEntitiesCatalog.ts

```diff
@@ -65,8 +71,31 @@ export class AuthorizedEntitiesCatalog implements EntitiesCatalog {
     return this.entitiesCatalog.entities(request);
   }
 
-  removeEntityByUid(uid: string): Promise<void> {
-    // TODO: Implement permissioning
+  async removeEntityByUid(
+    uid: string,
+    authorizationToken?: string,
+  ): Promise<void> {
+    const { entities } = await this.entitiesCatalog.entities({
+      filter: basicEntityFilter({ 'metadata.uid': uid }),
+    });
```

Comment:
I opted to make a db query eagerly, which allows us to provide the entityRef in the authorize request. This avoids the round trip for the call to the `analyze-conditions` endpoint (which, if called, would result in a db query below anyway).

The other option is to forgo the eager db query here and handle the conditional case below. This implementation could be advantageous if we think most policies will _not_ result in a conditional response for a delete request, which I don't think we can accurately conjecture.

Reply:
I lean toward omitting the `resourceRef` and using the conditions in the entities query. The intent of `resourceRef` described in #7761 was to support auth requests from the frontend or backends that don't "own" the resource being authorized, but it's worth discussing that more as we run into scenarios like this!

Reply:
Made the change, added corresponding conditional tests as well.